### PR TITLE
Improve docs with TypeDoc

### PR DESCRIPTION
**Pull request by Hugoo**
Mainly based on changes from Reto here, but skip Docusaurus and TypeDoc installation as discussed:
https://github.com/ERC725Alliance/erc725.js/pull/23

- I removed the docs folder because all the rst files / content are now within the source as TypeDoc comments
- I generated Markdown files with:

```
npx typedoc --out /docs-website/docs/tools/erc725js/technical-reference src/index.ts --includeVersion --hideInPageTOC true --sort visibility
```

and added the generated sources to docs portal: https://github.com/lukso-network/docs-website/commit/221c6b4647f747d51b6d87a19e2355af3451fff9

https://docs.lukso.tech/tools/erc725js/technical-reference/README

To do:
- make process less manual
- fix links (set right remote url when generating docs)
- add more examples

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,14 +2,12 @@
 
 This library allows for interfacing with ERC725Y compliant contract on an EVM using the [ERC725YJSONSchema](https://github.com/lukso-network/LIPs/blob/master/LSPs/LSP-2-ERC725YJSONSchema.md).
 
-For more information see [Documentation](https://erc725js.readthedocs.io/en/latest/).
+For more information see [Documentation](https://docs.lukso.tech/).
```
Comment (frozeman) [inline README.md:5]: this should link to the package docs, not just the generic docs

**Review by frozeman — APPROVED** [review]
How will this be imported to our docs, and how can we add custom pages, like a getting started?
Can you show me and example page?

**Comment by Hugoo** [comment]
We can add more .md files in this repo, they can be "picked up" by typedoc when we generate the docs.

Or we can create them in the docs repo directly, there is already a couple of files:

https://github.com/lukso-network/docs-website/tree/main/docs/tools/erc725js

The content of the "technical-reference" folder is the output of the typedoc script.

There is also a WIP for an article in this folder:

https://github.com/lukso-network/docs-website/commit/32d9b8ea352323468d498e9aab3a079bb947dbb2

**Comment by frozeman** [comment]
We should add them to this repository, in a docs folder. Ideally in a way that they can also be read directly on github.

There’s also some documentation in the files that were deleted, that should be transferred to markdown files first.

**Comment by rryter** [comment]
I merge this into `feature/docs`so that I can continue working on it.

**Pull request merged**